src/core/CMakeLists.txt

```diff
@@ -0,0 +1,13 @@
+cmake_minimum_required(VERSION 3.11)
+
+file(GLOB_RECURSE xournalpp_sources *.cpp *.h)
+set(xournalpp_include_dirs "${CMAKE_CURRENT_SOURCE_DIR}")
+
+add_library(xournalpp-core STATIC ${xournalpp_sources})
+target_compile_features(xournalpp-core PUBLIC ${PROJECT_CXX_FEATURES})
+target_link_libraries(xournalpp-core
+        PRIVATE xoj::util ${xournalpp_LDFLAGS}
+        PUBLIC xoj::defaults xoj::external_modules std::filesystem cxx17
+        )
+target_include_directories(xournalpp-core PUBLIC ${xournalpp_include_dirs})
+add_library(xoj::core ALIAS xournalpp-core)
```

Comment:
Why make this an alias if you could just use the actual name throughout the whole file?

Reply:
Doesn't have a higher meaning. But it moves our targets into a namespace and makes them read only. They are also visible everywhere, even in top level CMakeLists.txt.